app.ts

```diff
@@ -1,96 +1,52 @@
-  const timeStamp: Quad = member.quads.find(
-    (quadObj) => quadObj.predicate.value === process.env.LDES_RELATION_PATH
-  );
-  return timeStamp.object;
-}
-
-function extractBaseResourceUri(member): NamedNode {
-  const baseResourceMatches = member.quads.filter((quadObj) =>
-    quadObj.predicate.equals(purl("isVersionOf"))
-  );
-  if (baseResourceMatches && baseResourceMatches.length) {
-    return baseResourceMatches[0].object;
-  }
-  return;
-}
-
-async function processMember(member) {
-  console.log("process start");
-  const quadsToAdd: Quad[] = member.quads.filter(
-    (quadObj) => quadObj.subject.value === member.id.value
-  );
-  const quadsToRemove: Quad[] = [];
-  const baseResourceUri = extractBaseResourceUri(member);
-  if (baseResourceUri && process.env.REPLACE_VERSIONS) {
-    quadsToRemove.push(
-      quad(variable("s"), purl("isVersionOf"), baseResourceUri)
-    );
-    quadsToAdd.forEach((quadObj, i) => {
-      quadsToRemove.push(
-        quad(variable("s"), quadObj.predicate, variable(`o${i}`))
-      );
-    });
-  }
-
-  await executeDeleteInsertQuery(quadsToRemove, quadsToAdd);
-  await processTimeStamp(member);
-  console.log("process end");
-}
-
-async function processTimeStamp(member) {
-  try {
-    let timestamp = extractTimeStamp(member);
-    if (timestamp) {
-      await replaceEndTime(timestamp);
-    }
-  } catch (e) {
-    throw e;
-  }
-}
-
-async function main() {
-  try {
-    const endTime = await getEndTime();
-    console.log("Start time: ", endTime);
-    let url = process.env.LDES_ENDPOINT_VIEW;
-    let options = {
-      pollingInterval: parseInt(process.env.LDES_POLLING_INTERVAL), // millis
-      representation: "Quads", //Object or Quads
-      fromTime: endTime,
-      mimeType: "application/ld+json",
-      emitMemberOnce: true,
-    };
-    let LDESClient = new newEngine();
-
-    let eventstreamSync = LDESClient.createReadStream(url, options);
-    eventstreamSync.on("data", (member) => {
-      const baseResourceUri = extractBaseResourceUri(member);
-      if (baseResourceUri) {
-        UPDATE_QUEUE.push(() => processMember(member));
-      }
-    });
-  } catch (e) {
-    console.error(e);
-  }
+import * as RDF from "rdf-js";
+import Consumer, { Member } from "./consumer";
+import { convertBlankNodes, extractBaseResourceUri } from "./utils";
+import { CronJob } from 'cron';
+import { CRON_PATTERN, LDES_ENDPOINT_VIEW, LDES_POLLING_INTERVAL, REPLACE_VERSIONS } from "./config";
+const { quad, variable } = DataFactory;
+
+async function processMember(member: Member) {
+	const quadsToAdd: RDF.Quad[] = member.quads;
+	const quadsToRemove: RDF.Quad[] = [];
+	const baseResourceUri = extractBaseResourceUri(member);
+	if (baseResourceUri && REPLACE_VERSIONS) {
+		quadsToRemove.push(
+			quad(variable("s"), PURL("isVersionOf"), baseResourceUri)
+		);
+    quadsToRemove.push(quad(variable("s"), variable("p"), variable("o")))
+	}
+	await executeDeleteInsertQuery(quadsToRemove, quadsToAdd);
```

Comment:
I am not sure how we should use this callback function to execute custom storage rules?

Reply:
my idea ideally would be: you mount a custom file in /config with specifc signature, so you could also I dunno do some other calls.
What we have as default behaviour is nice btw. But in the context of ipd-lpdc, we'll have to write a consumer that pushes a json ld call

Reply:
The idea would be that we extract the Consumer class to a seperate library, which can also be used for a service which pushes json calls.

Reply:
ah ok